File: DoItTest.Domain/Tests/TestItems/AnswerOptions/ComparisonAnswerOption.cs
namespace DoItTest.Domain.Tests.TestItems.AnswerOptions
{
    public class ComparisonAnswerOption : AnswerOption
    {
        public Guid? GroupId { get; private set; }
        public String? GroupName { get; private set; }
        public String Title { get;  }

        public ComparisonAnswerOption(Guid id, Guid testItemId, TestItemType type, Guid? groupId, String? groupName, String title) : base(id, testItemId, type)
        {
            GroupId = groupId;
            GroupName = groupName;
            Title = title;
        }

        public void ClearGroup()
        {
            GroupId = null;
            GroupName = null;
        }
    }
}

File: DoItTest/ClientApp/src/domain/tests/items/answerOption/comparisonAnswerOption.ts
import { TestItemType } from "../testItemType"
import { AnswerOption } from "./answerOption"

export class ComparisonAnswerOption extends AnswerOption {
    constructor(
        public id: string,
        public testItemId: string,
        public type: TestItemType,
        public groupId: string | null,
        public title: string
    ) {
        super(id, testItemId, type)
    }
}

export function mapComparisonAnswerOption(value: any): ComparisonAnswerOption {
    return new ComparisonAnswerOption(value.id, value.testItemId, value.type, value.groupId, value.title)
}

export function mapComparisonAnswerOptions(value: any[]): ComparisonAnswerOption[] {
    return value.map(mapComparisonAnswerOption)
}